configs/repos/ufs/packages/wgrib2/package.py

```diff
@@ -0,0 +1,129 @@
+    variant('ipolates', default='3', 
+        description='Link in IPOLATES library to interpolate to new grids (0=OFF, 1=ip, 3=ip2)', 
+        values=('0', '1', '3'))
+    variant('spectral', default=False, description='Spectral interpolation in -new_grid')
+    variant('fortran_api', default=True, description='Make wgrib2api which allows fortran code to read/write grib2')
+    variant('mysql', default=False, description='Link in interface to MySQL to write to mysql database')
+    variant('udf', default=False, description='Add commands for user-defined functions and shell commands')
+    variant('regex', default=True, description='Use regular expression library (POSIX-2')
+    variant('tigge', default=True, description='Ability for TIGGE-like variable names')
+    variant('proj4', default=False, description='The proj4 library is used to confirm that the gctpc code is working correctly')
+    variant('aec', default=True, description='Enable use of the libaec library for packing with GRIB2 template')
+    variant('g2c', default=False, description='include NCEP g2clib (mainly for testing purposes)')
+    variant('disable_timezone', default=False, description='Some machines do not support timezones')
+    variant('disable_alarm', default=False, description='Some machines do not support alarm(..) (not POSIX-1, IEEE Std 1003.1) use the alarm to terminate wgrib2 after N seconds')
+    variant('png', default=True, description='PNG encoding')
+    variant('jasper', default=True, description='JPEG compression using Jasper')
+    variant('openmp', default=True, description='OpenMP parallelization')
+    variant('wmo_validation', default=False, description='WMO validation')
+
+    variant_map = {
+        'netcdf3': 'USE_NETCDF3',
+        'netcdf4': 'USE_NETCDF4',
+        'spectral': 'USE_SPECTRAL',
+        'mysql': 'USE_MYSQL',
+        'udf': 'USE_UDF',
+        'regex': 'USE_REGEX',
+        'tigge': 'USE_TIGGE',
+        'proj4': 'USE_PROJ4',
+        'aec': 'USE_AEC',
+        'g2c': 'USE_G2CLIB',
+        'png': 'USE_PNG',
+        'jasper': 'USE_JASPER',
+        'openmp': 'USE_OPENMP',
+        'wmo_validation': 'USE_WMO_VALIDATION',
+        'ipolates': 'USE_IPOLATES',
+        'disable_timezone': 'DISABLE_TIMEZONE',
+        'disable_alarm': 'DISABLE_ALARM',
+        'fortran_api': 'MAKE_FTN_API'
+    }
+           
+    # Disable parallel build
+    parallel = False
+
+    def url_for_version(self, version):
+        url = "https://www.ftp.cpc.ncep.noaa.gov/wd51we/wgrib2/wgrib2.tgz.v{}"
+        return url.format(version)
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter('makefile')
+
+        for variant_name, makefile_option in self.variant_map.items():
+            value = int(spec.variants[variant_name].value)
+            makefile.filter(r'^%s=.*' % makefile_option, '{}={}'.format(makefile_option, value))
+
+    def build(self, spec, prefix):
```

Comment:
I see you are installing everything as part of the build. I had to make a similar choice for a new package `shumlib`, but in this case it was possible to "install" during the build phase into a subdirectory "build" and then in the real install phase move things over using `install_tree`.

I thought this would reduce the risk of a half broken installation. For example, if building `wgrib2` works, but the fortran api fails, will the wgrib2 binary already be in place - or will spack roll back the install command? Do you know?

Reply:
The thing is that wgrib2 is built in two phases for the executable and the library, with a `make clean` in between which deletes the executable. I think I could work around that by saving the executable before cleaning.

Reply:
Fixed. I move the products into a temporary directory and use `install_tree` to install.